src/main/java/net/jodah/failsafe/Fallback.java

```diff
@@ -136,7 +136,10 @@ private Fallback(CheckedFunction<ExecutionAttemptedEvent, R> fallback, boolean a
   }
 
   public R apply(R result, Throwable failure, ExecutionContext context) throws Exception {
-    return fallback.apply(new ExecutionAttemptedEvent<R>(result, failure, context));
+    if (failure instanceof Error) {
```

Comment:
Fallback should not blindly handle errors - in fact I think it should bias toward throwing them.

However, we could in theory apply Failure policies here too to determine whether the Error should be handled or not (that'd probably require some refactoring).

Alternatively, we could accept a set of handled `Error` classes, this would create some duplication between the retry policy and the fallback, but would require fewer code changes.

I personally think this could be left as is, because if you really want to catch errors and have a fallback you can still wrap in a `try-catch` block.